cpp/src/arrow/filesystem/s3_internal.h

```diff
@@ -88,6 +88,84 @@ inline bool IsAlreadyExists(const Aws::Client::AWSError<Aws::S3::S3Errors>& erro
+      return "INCOMPLETE_SIGNATURE";
+    case Aws::S3::S3Errors::INTERNAL_FAILURE:
+      return "INTERNAL_FAILURE";
+    case Aws::S3::S3Errors::INVALID_ACTION:
+      return "INVALID_ACTION";
+    case Aws::S3::S3Errors::INVALID_CLIENT_TOKEN_ID:
+      return "INVALID_CLIENT_TOKEN_ID";
+    case Aws::S3::S3Errors::INVALID_PARAMETER_COMBINATION:
+      return "INVALID_PARAMETER_COMBINATION";
+    case Aws::S3::S3Errors::INVALID_QUERY_PARAMETER:
+      return "INVALID_QUERY_PARAMETER";
+    case Aws::S3::S3Errors::INVALID_PARAMETER_VALUE:
+      return "INVALID_PARAMETER_VALUE";
+    case Aws::S3::S3Errors::MISSING_ACTION:
+      return "MISSING_ACTION";
+    case Aws::S3::S3Errors::MISSING_AUTHENTICATION_TOKEN:
+      return "MISSING_AUTHENTICATION_TOKEN";
+    case Aws::S3::S3Errors::MISSING_PARAMETER:
+      return "MISSING_PARAMETER";
+    case Aws::S3::S3Errors::OPT_IN_REQUIRED:
+      return "OPT_IN_REQUIRED";
+    case Aws::S3::S3Errors::REQUEST_EXPIRED:
+      return "REQUEST_EXPIRED";
+    case Aws::S3::S3Errors::SERVICE_UNAVAILABLE:
+      return "SERVICE_UNAVAILABLE";
+    case Aws::S3::S3Errors::THROTTLING:
+      return "THROTTLING";
+    case Aws::S3::S3Errors::VALIDATION:
+      return "VALIDATION";
+    case Aws::S3::S3Errors::ACCESS_DENIED:
+      return "ACCESS_DENIED";
+    case Aws::S3::S3Errors::RESOURCE_NOT_FOUND:
+      return "RESOURCE_NOT_FOUND";
+    case Aws::S3::S3Errors::UNRECOGNIZED_CLIENT:
+      return "UNRECOGNIZED_CLIENT";
+    case Aws::S3::S3Errors::MALFORMED_QUERY_STRING:
+      return "MALFORMED_QUERY_STRING";
+    case Aws::S3::S3Errors::SLOW_DOWN:
+      return "SLOW_DOWN";
+    case Aws::S3::S3Errors::REQUEST_TIME_TOO_SKEWED:
+      return "REQUEST_TIME_TOO_SKEWED";
+    case Aws::S3::S3Errors::INVALID_SIGNATURE:
+      return "INVALID_SIGNATURE";
+    case Aws::S3::S3Errors::SIGNATURE_DOES_NOT_MATCH:
+      return "SIGNATURE_DOES_NOT_MATCH";
+    case Aws::S3::S3Errors::INVALID_ACCESS_KEY_ID:
+      return "INVALID_ACCESS_KEY_ID";
+    case Aws::S3::S3Errors::REQUEST_TIMEOUT:
+      return "REQUEST_TIMEOUT";
+    case Aws::S3::S3Errors::NETWORK_CONNECTION:
+      return "NETWORK_CONNECTION";
+    case Aws::S3::S3Errors::UNKNOWN:
+      return "UNKNOWN";
+    case Aws::S3::S3Errors::BUCKET_ALREADY_EXISTS:
+      return "BUCKET_ALREADY_EXISTS";
+    case Aws::S3::S3Errors::BUCKET_ALREADY_OWNED_BY_YOU:
+      return "BUCKET_ALREADY_OWNED_BY_YOU";
+    case Aws::S3::S3Errors::INVALID_OBJECT_STATE:
+      return "INVALID_OBJECT_STATE";
+    case Aws::S3::S3Errors::NO_SUCH_BUCKET:
+      return "NO_SUCH_BUCKET";
+    case Aws::S3::S3Errors::NO_SUCH_KEY:
+      return "NO_SUCH_KEY";
+    case Aws::S3::S3Errors::NO_SUCH_UPLOAD:
+      return "NO_SUCH_UPLOAD";
+    case Aws::S3::S3Errors::OBJECT_ALREADY_IN_ACTIVE_TIER:
+      return "OBJECT_ALREADY_IN_ACTIVE_TIER";
+    case Aws::S3::S3Errors::OBJECT_NOT_IN_ACTIVE_TIER:
+      return "OBJECT_NOT_IN_ACTIVE_TIER";
+    default:
+      DCHECK(false);
```

Comment:
Not sure if this `DCHECK` is a good idea -- if we fix another problem with the AWS SDK that I found upstream https://github.com/aws/aws-sdk-cpp/pull/2062, I already know this will break. Same if new errors get added in the future. Better to just give the `[code <error code>]` answer if a newer SDK is linked with an older arrow?

Reply:
No, it's definitely not a good idea, can you remove it?

Reply:
Thanks, I fixed it :)